pkg/pushsync/pushsync.go

```diff
@@ -546,56 +553,45 @@ func (ps *PushSync) warmedUp() bool {
 
 type peerSkipList struct {
 	sync.Mutex
-	chunks         map[string]struct{}
-	skipExpiration map[string]time.Time
+
+	// key is chunk address, value is map of peer address to expiration
+	skip map[string]map[string]time.Time
 }
 
 func newPeerSkipList() *peerSkipList {
 	return &peerSkipList{
-		chunks:         make(map[string]struct{}),
-		skipExpiration: make(map[string]time.Time),
+		skip: make(map[string]map[string]time.Time),
 	}
 }
 
 func (l *peerSkipList) Add(peer, chunk swarm.Address, expire time.Duration) {
```

Comment:
For better clarity I would prefer chunk as the first parameter

Reply:
not sure WDYM. the map is a list of chunks for which you'd like to skip peers for a certain amount of time. if a forwarding request for a certain chunk comes again, you'd have to traverse the entire map to get the list of peers to skip for that chunk

Reply:
I just mean the order of the parameters. As the base map is keyed by chunk, it would be better to have that as first parameter.

Reply:
aha yes I see what you mean now. Sorry I missed which line you were commenting on :facepalm: